### PR TITLE
Update github actions

### PR DESCRIPTION
**Pull request by newAM**
* `actions/checkout@v2` -> `actions/checkout@v3`
  * v2 uses node12 which is deprecated, and soon to be unsupported by github actions
* `actions-rs/toolchain` -> `dtolnay/rust-toolchain`
  * `actions-rs` is unmaintained: https://github.com/actions-rs/toolchain/issues/216

### CODE REVIEW

**Comment by newAM** [comment]
This is untested, it may need a fix or two after the workflows are approved and run :)

**Comment by newAM** [comment]
Huh, I think that clippy failure is just github actions having a bad day.  The http failure 429 is "too many requests", have not seen that one before.

File: .github/workflows/ci.yml
Comment (ryan-summers) [inline .github/workflows/ci.yml:49]: Can we pin tool chain to a version? I don't like running off master in case it causes random breakage personally

**Review by ryan-summers — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml
Comment (newAM) [reply .github/workflows/ci.yml:49]: dtolnay implemented it such that the version tags correspond to the toolchain version.  `master` is used when the toolchain version is parametric.

I'm not too concerned about stability since dtolnay has a good record of API stability from serde and proc macros, and I have not had any issues in ~1yr of using it for my own repos.

That all said, I'm happy to find something else if you disagree :)

**Review by ryan-summers — APPROVED** [review]
LGTM, we can revisit tags if it's a problem. Better than using deprecated stuff :)

**Pull request merged**
